modules/internal/String.chpl

```diff
@@ -1208,12 +1290,20 @@ module String {
 
       on __primitive("chpl_on_locale_num",
                      chpl_buildLocaleID(this.locale_id, c_sublocid_any)) {
-        for i in 0..#this.len {
-          const char = buff[i];
-          if char <= 0x1f  || char == 0x7f {
+        var i = 0;
+        while i < this.len {
+          var codepoint: int(32);
+          var nbytes: c_int;
+          var multibytes = (this.buff + i): c_string;
+          var maxbytes = (this.len - i): ssize_t;
+          if maxbytes > 4 then
```

Comment:
here... Now that I'm mentioning it.. why don't these use the string's codepoint iterator uchars() ?

Reply:
Good point.  Most of this code evolved independently of the iterator.  Now that I have them in the same branch together, I can use `uchars()`.  That makes the code easier to read.